ldap3/protocol/rfc4512.py

```diff
@@ -446,7 +446,7 @@ def from_definition(cls, definitions):
                 else:
                     raise LDAPSchemaError('unknown schema definition class')
 
-                splitted = re.split('( NAME | DESC | OBSOLETE| X-| E-' + pattern + ')', object_definition[1:-1])
+                splitted = re.split('( NAME | DESC | OBSOLETE| X-| E-' + pattern + ')', str(object_definition[1:-1]))
```

Comment:
`str` means different things on Python 2 and 3, so this is likely to be non-portable. Start with a test case?